### PR TITLE
Inherit defaults from Fabric8 kubernetes client defaults

### LINKED ISSUE #43474: Improper defaults in KubernetesClientBuildConfig
### Describe the bug

Not strictly a bug but rather an annoyance: the default values that are provided in `KubernetesClientBuildConfig` are not up-to-date which leads to behavioral changes between a default configured client and one that's injected by Quarkus. 

### Expected behavior

The defaults should be consistent with what users would expect when they're not using Quarkus

### Actual behavior

The defaults do not match the current implementation anymore.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

From a maintenance perspective, providing such defaults opens the door to the current issue where defaults are not in synch. The implementation is already set up to inherit defaults from the implementation so we should use that instead.

### PR DESCRIPTION
**Pull request by metacosm**
Fixes #43474

Signed-off-by: Chris Laprun <claprun@redhat.com>

### CODE REVIEW

File: extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientBuildConfig.java [hunk 1]
```diff
@@ -93,39 +93,33 @@ public interface KubernetesClientBuildConfig {
...
-    Duration watchReconnectInterval();
+    Optional<Duration> watchReconnectInterval();
 
     /**
      * Maximum reconnect attempts in case of watch failure
      * By default there is no limit to the number of reconnect attempts
      */
-    @WithDefault("-1") // default lifted from Kubernetes Client
-    int watchReconnectLimit();
+    Optional<Integer> watchReconnectLimit();
```
Comment (gastaldi) [inline extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientBuildConfig.java:102]:
```suggestion
    OptionalInt watchReconnectLimit();
```

**Comment by metacosm** [comment]
I haven't considered backwards compatibility, not sure this is something that needs to be taken into account here.

**Review by gastaldi — APPROVED** [review]

**Comment by manusa** [comment]
I'd vote to backport this PR too since it's addressing a big issue.
Quarkus is overriding some of the Kubernetes Client defaults which might drastically change the expected client behavior.
This is especially notable for the retry functionality which is effectively disabled by the overridden default.

**Comment by gsmet** [comment]
@metacosm @manusa could you prepare a note for the migration guide as to which default values were changed to what? Because while I understand the new defaults are better, people might still rely on the old defaults for some use case: we need to document the change.

You can post a comment here and I'll post it to the appropriate migration guide.

**Comment by metacosm** [comment]
The previous defaults appeared in `KubernetesClientBuildConfig` in the `@Default` annotations. The new ones cannot really be documented once and for all since the whole point of this PR is to follow whatever the client does as it changes over time.

**Comment by manusa** [comment]
I can try to compile the values from before and after and create a table with the changed ones.
Maybe we can clarify in the note that starting from this version the values will default to whatever you find in https://github.com/fabric8io/kubernetes-client#configuring-the-client

**Comment by manusa** [comment]
Hi @gsmet 

I've compiled all of the configuration options, their old defaults, and the new defaults.
The following snippet contains some markdown that can be adapted to fit in the migration guide:

```md
Starting from the xxxx version, the default values for the Kubernetes Client extension configuration have changed.
In the past, Quarkus had explicit values for some of the Kubernetes Client configuration options.
This was not ideal since some of these default settings were different to those provided by the client library which led to some confusion.
To address this, we have removed the explicit default values from the Kubernetes Client extension configuration and now rely on the client library defaults.

The following table contains all the configuration option entries, the previous default value, and the new default value (note that many of the values remain unchanged and others don't have a default value):

| Configuration Option          | Previous Default Value | New Default Value |
|-------------------------------|------------------------|-------------------|
| `watchReconnectInterval`      | `PT1S`                 | `PT1S`            |
| `watchReconnectLimit`         | `-1` (no limit)        | `-1` (no limit)   |
| `connectionTimeout`           | `PT10S`                | `PT10S`           |
| `requestTimeout`              | `PT10S`                | `PT10S`           |
| `requestRetryBackoffLimit`    | `0` (no retry)         | `10`              |
| `requestRetryBackoffInterval` | `PT1S`                 | `PT0.1S`          |
| `trustCerts`                  | n/a                    | `false`           |
| `apiServerUrl`                | n/a                    | n/a               |
| `namespace`                   | n/a                    | n/a               |
| `caCertFile`                  | n/a                    | n/a               |
| `caCertData`                  | n/a                    | n/a               |
| `clientCertFile`              | n/a                    | n/a               |
| `clientCertData`              | n/a                    | n/a               |
| `clientKeyFile`               | n/a                    | n/a               |
| `clientKeyData`               | n/a                    | n/a               |
| `clientKeyAlgo`               | n/a                    | `RSA`             |
| `clientKeyPassphrase`         | n/a                    | n/a               |
| `username`                    | n/a                    | n/a               |
| `password`                    | n/a                    | n/a               |
| `token`                       | n/a                    | n/a               |
| `httpProxy`                   | n/a                    | n/a               |
| `httpsProxy`                  | n/a                    | n/a               |
| `proxyUsername`               | n/a                    | n/a               |
| `proxyPassword`               | n/a                    | n/a               |
| `noProxy`                     | n/a                    | n/a               |
```

**Pull request merged**